codespan-reporting/src/term/views.rs

```diff
@@ -100,14 +100,18 @@ where
                 .find(|labeled_file| label.file_id == labeled_file.file_id)
             {
                 Some(labeled_file) => {
-                    if labeled_file.start > label.range.start {
+                    // another diagnostic also referenced this file
+                    if labeled_file.start > label.range.start && label.style == LabelStyle::Primary
+                    {
+                        // this label indicates an earlier start
                         labeled_file.start = label.range.start;
                         labeled_file.location =
                             files.location(label.file_id, label.range.start).unwrap();
                     }
                     labeled_file
                 }
                 None => {
+                    // no other diagnostic referenced this file yet
                     labeled_files.push(LabeledFile {
                         file_id: label.file_id,
                         start: label.range.start,
```

Comment:
To make the solution truly universal, it would have to be made sure that this case is first triggered by a primary label.

Or LabeledFile::location could also be an `Optional` so it is only set to `Some` when a primary label is found and the positions of secondary labels would be ignored. This would add an error condition.

Reply:
I wonder if instead we could have a `max_label_style: LabelStyle` field on `LabeledFile`? This would let us fall-back to the first secondary label, and then as soon as we see the first primary label we'd be able to switch to using that as our start position.